File: soaint-sgd-enterprise-artifacts/canonical-api-contract/src/main/java/co/com/soaint/foundation/canonical/correspondencia/ComunicacionOficialDTO.java
package co.com.soaint.foundation.canonical.correspondencia;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.xml.bind.annotation.XmlRootElement;
import java.io.Serializable;
import java.util.List;

/**
 * ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 * Soaint Generic Artifact
 * Created:12-Jun-2017
 * Author: esanchez
 * Type: JAVA class Artifact
 * Purpose: DTO - Model Artifact
 * ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 */

@Data
@Builder(builderMethodName = "newInstance")
@NoArgsConstructor
@AllArgsConstructor
@XmlRootElement(namespace = "http://soaint.com/domain-artifacts/correspondencia/1.0.0")
public class ComunicacionOficialDTO implements Serializable {

    private static final long serialVersionUID = 1L;

    private CorrespondenciaDTO correspondencia;
    private List<AgenteDTO> agenteList;
    private PpdDocumentoDTO ppdDocumento;
    private List<AnexoDTO> anexoList;
    private List<ReferidoDTO> referidoList;
}

File: soaint-sgd-ui/soaint-sgd-web/src/app/domain/ComunicacionOficialDTO.ts
import {ContactoDTO} from './ContactoDTO';
import {ReferidoDTO} from './ReferidoDTO';
import {AnexoDTO} from './AnexoDTO';
import {DocumentoDTO} from './DocumentoDTO';
import {AgentDTO} from './AgentDTO';
import {CorrespondenciaDTO} from './correspondenciaDTO';


export interface ComunicacionOficialDTO {
  correspondencia?: CorrespondenciaDTO,
  agenteList?: Array<AgentDTO>,
  ppdDocumento?: DocumentoDTO,
  anexoList?: Array<AnexoDTO>
  referidoList?: Array<ReferidoDTO>
  datosContactoList?: Array<ContactoDTO>
}
